src/components/PuzzleGrid/index.jsx

```diff
@@ -0,0 +1,127 @@
+const PuzzleGrid = () => {
+  const [currentPuzzleState, setCurrentPuzzleState] = useState([]);
+  // selectedTile state shows if there is a tile selected and which one is selected if any
+  const [selectedTile, setSelectedTile] = useState(null);
+  // if unsolved it displays the tiles and when solved it displays the full image
+  const [isSolved, setIsSolved] = useState(false);
+
+  const tilesGenerator = () => {
+    const newTilesArray = [];
+    // for loop to generate tiles where tiles are puzzle pieces
+    for (let tileId = 0; tileId < NUM_OF_TILES; tileId++) {
+      let tileTopStyling = tileId * 100;
+      // value of selected is used for styling purposes where the selected tile should be highlighted
+      // value of top is used for fitting the image inside the tile
+      let newTile = { id: tileId, selected: false, top: tileTopStyling };
+      newTilesArray.push(newTile);
+    }
+
+    const randomizedTilesArray = shuffle(newTilesArray);
+    // Set the randomized array as generated tiles
+    setCurrentPuzzleState(randomizedTilesArray);
+  };
+
+  useEffect(() => {
+    tilesGenerator();
+  }, []);
+
+  useEffect(() => {
+    checkSolution();
+  }, [currentPuzzleState]);
+
+  const handleSelect = (clickedTileId) => {
+    // if selectedTile's value is not equal to null (meaning there a tile seleted already)
+    // then when another tile is clicked swap both tiles' places
+    // else set the selectedTile to the id of the clicked tile
+    if (selectedTile !== null) {
+      const prevId = selectedTile;
+      swapTiles(prevId, clickedTileId);
+    } else {
+      setSelectedTile(clickedTileId);
+      const clickedTileIndex = currentPuzzleState.findIndex(
+        (tile) => tile.id === clickedTileId
+      );
+      // set the select value of the tile to true
+      currentPuzzleState[clickedTileIndex] = {
+        ...currentPuzzleState[clickedTileIndex],
+        selected: true,
+      };
+    }
+  };
+
+  const swapTiles = (id0, id1) => {
+    let newTiles = [...currentPuzzleState];
+    let index0 = currentPuzzleState.findIndex((tile) => tile.id === id0);
+    let index1 = currentPuzzleState.findIndex((tile) => tile.id === id1);
+    let tile0 = { ...currentPuzzleState[index0] };
+    let tile1 = { ...currentPuzzleState[index1] };
+    newTiles[index0] = tile1;
+    newTiles[index1] = tile0;
+    newTiles.forEach((tile) => (tile.selected = false));
+    setCurrentPuzzleState(newTiles);
+    setSelectedTile(null);
+  };
+
+  const checkSolution = () => {
+    // tiles are saved in an array indexed from 0 to n (6) and displayed in that order
+    // first condition checks if tile's id is equal to its index in the array and returns false/true
+    // since initial acc value is true, second condition checks if prev value is true and keeps its value
+    const isCorrect = currentPuzzleState.reduce((acc, curr, index) => {
+      if (curr.id !== index || acc === false) {
```

Comment:
`curr.id` and `index `are not descriptive. Are you comparing current position and correct position? If yes you should name them accordingly.

Reply:
PTAL